liquidctl/driver/nzxt_smart_device.py

```diff
@@ -163,29 +217,112 @@ def get_status(self, **kwargs):
+            # Get configuration information from device
+            wmsg = [0x10, 0x01, 0x00, 0x00, 0x00, 0x00]
+            LOGGER.info('Issuing command 0x10 0x01 to get firmware info')
+            self._write(wmsg)
+            wmsg = [0x20, 0x03, 0x00, 0x00, 0x00, 0x00]
+            LOGGER.info('Issuing command 0x20 0x03 to get lighting info')
+            self._write(wmsg)
+            #wmsg = [0x60, 0x03, 0x00, 0x00, 0x00, 0x00]
+            #LOGGER.info('Issuing command 0x20 0x03 to get lighting info')
+            #self._write(wmsg)
+            # After issuing the above 2 commands, we will get a series of replies that
+            # will include everything we want to extract and display to the user.
+            # It may take 10 or 12 reply messages before we get all of the expected replies.
+            for x in range(12):     # we may not get reply right away, so try up to 12 times
+                if numValidRepliesReceived == 4:
+                    break   
+                msg = self.device.read(_READ_LENGTH_V2)
+                LOGGER.debug('received %s', ' '.join(format(i, '02x') for i in msg))
+                if msg1101Reply == False and msg[0] == 0x11 and msg[1] == 0x01:  # the correct reply must start with 0x11, 0x01
+                    fw = '{}.{}.{}'.format(msg[0x11], msg[0x12], msg[0x13])
+                    status.append(('Firmware:', fw, ''))
+                    numValidRepliesReceived += 1 
+                    msg1101Reply = True
+                    continue
+                if msg2103Reply == False and msg[0] == 0x21 and msg[1] == 0x03:  # the correct reply must start with 0x21, 0x03
+                    ledStripCount = 0
+                    ledUnderglowCount = 0
+                    ledCableCombCount = 0
+                    ledAerFan120Count = 0
+                    ledAerFan140Count = 0
+                    for ledIter in range(15):
+                        ledCode = msg[0x0f + ledIter]
+                        if ledCode == 0x04:
+                            ledStripCount += 1
+                            ledName = '  LED Strip #{}'.format(ledStripCount)
+                        elif ledCode == 0x08:
+                            ledCableCombCount += 1
+                            ledName = ' Cable Comb #{}'.format(ledCableCombCount)
+                        elif ledCode == 0x0a:
+                            ledUnderglowCount += 1
+                            ledName = '  Underglow #{}'.format(ledUnderglowCount)
+                        elif ledCode == 0x0b:
+                            ledAerFan120Count += 1
+                            ledName = 'AER Fan 120 #{}'.format(ledAerFan120Count)
+                        elif ledCode == 0x0c:
+                            ledAerFan140Count += 1
+                            ledName = 'AER Fan 140 #{}'.format(ledAerFan140Count)
+                        if ledCode != 0:
+                            status.append(('    HUE2:', ledName, ''))
```

Comment:
I like your general idea, but I suggest something more in the lines of:

```
# liquidctl status
[...]
LED 1 accessory 1    Aer RGB 2 120 mm
LED 1 accessory 2    Aer RGB 2 120 mm
LED 1 accessory 3    Aer RGB 2 140 mm
LED 1 accessory 4    HUE 2 Cable Comb
LED 2 accessory 1    Hue+ Strip
LED 2 accessory 2    Hue+ Strip
[...]
```

For the second accessory, this would be equivalent to the following tuple:

```python
('LED 1 accessory 2', 'Aer RGB 2 120mm', '')
```

Also, don't align the output of the CLI within this function.  `get_status` can be called from other code and these days the printing function in `liquidctl.cli` can also do some additional formatting.

Reply:
Done. Here's a sample output:
```
Device 1, NZXT Smart Device V2
Fan 2 duty                    40  %  
Fan 2 speed                  875  rpm
Firmware:                  1.5.0     
LED 1 accessory 1     HUE 2 LED Strip     
LED 1 accessory 2     HUE 2 Underglow     
LED 1 accessory 3     HUE 2 Underglow     
LED 2 accessory 1     AER RGB 2 140 mm     
LED 2 accessory 2     AER RGB 2 140 mm     
LED 2 accessory 3     AER RGB 2 140 mm     
LED 2 accessory 4     AER RGB 2 120 mm     
Noise level                   66  dB
```

Reply:
[Automatic column sizing](https://github.com/jonasmalacofilho/liquidctl/blob/db9ca13eab125e184de38e649e6e57601f2c67fc/liquidctl/cli.py#L177-L201) is not working as well as I thought.  I'll take a look.

Leaving it open for now...

Reply:
Actually, I can't locally reproduce the column sizing bug.

Any change you're applying the patch on top of 1.2.0, instead of the current master branch?

Reply:
Oh yes, I tested with an earlier version of liquidctl. Will install latest files and see if the columns line up.

Reply:
Ok, that should explain it.  Marking this as resolved.